deepchem/utils/save.py

```diff
@@ -104,6 +106,68 @@ def load_csv_files(filenames, shard_size=None, verbose=True):
         yield df
 
 
+def seq_one_hot_encode(sequences):
+  """One hot encodes list of genomic sequences.
+
+  Sequences encoded have shape (N_sequences, 4, sequence_length, 1).
+  Here 4 is for the 4 basepairs (ACGT) present in genomic sequences.
+  These sequences will be processed as images with one color channel.
+
+  Parameters
+  ----------
+  sequences: np.ndarray 
+    Array of genetic sequences 
+  """
+  sequence_length = len(sequences[0])
+  # depends on Python version
+  integer_type = np.int32
+  # The label encoder is given characters for ACGTN
+  label_encoder = LabelEncoder().fit(np.array(('ACGTN',)).view(integer_type))
+  # These are transformed in 0, 1, 2, 3, 4 in input sequence
+  integer_array = []
+  # TODO(rbharath): Unlike the DRAGONN implementation from which this
+  # was ported, I couldn't transform the "ACGT..." strings into
+  # integers all at once. Had to do one at a time. Might be worth
+  # figuring out what's going on under the hood.
+  for sequence in sequences:
+    integer_seq = label_encoder.transform(
+        np.array((sequence,)).view(integer_type))
+    integer_array.append(integer_seq)
+  integer_array = np.concatenate(integer_array)
+  integer_array = integer_array.reshape(len(sequences), sequence_length)
```

Comment:
I'm suspicous of this reshape -- can we get a test?

Reply:
Good catch. Turns out that there was an error case when input sequences had different lengths. Raise a `ValueError` for that now, and added a couple more tests.